users/app.py

```diff
@@ -54,22 +54,54 @@ def add_update_user():
     return jsonify(user_object), (201 if added_new_user else 200)
 
 
+def upsert_user_in_db(user_object, users_collection):
+    """Updates or inserts the user object into users_collection.
+
+    Sets/overwrites default values as needed.
+
+    user_object should look like
+        {"user_id": foobar,
+         "name": Foo Bar}
+
+    Documents in users_collection should look like
+        {"user_id": foobar,
+         "name": Foo Bar,
+         "is_organizer": False}
+
+    Returns:
+        dict: The upserted object as it is in the db.
+
+    Raises:
+        AttributeError: if user_object is malformatted.
+    """
+    if user_object.keys() < {"user_id", "name"}:
+        raise AttributeError("missing user_id or name")
+    # insert is_organizer field (default False)
+    user_object["is_organizer"] = False
+    # upsert user in db
+    users_collection.update_one(
+        {"user_id": user_object["user_id"]},
+        {"$set": user_object},
+        upsert=True)
+    return users_collection.find_one({"user_id": user_object["user_id"]})
```

Comment:
Looking at the Mongo API, it looks like `update_one` returns an [UpdateResult](https://api.mongodb.com/python/current/api/pymongo/results.html#pymongo.results.UpdateResult), which inculdes an `upserted_id` and a `raw_result`, which is "the raw result document returned by the server".

Reply:
The raw_result field confusingly doesn't actually give back the whole document that was updated. Instead, it looks like
```python
{'connectionId': 0, 'err': None, 'n': 1, 'nModified': 0, 'ok': 1, 'updatedExisting': False, 'upserted': ObjectId('5d2d025007...1bb77ad4')}
```
It seems like it only gives some information about the updating operation. Since updating in Mongo allows you to partially update a document, I'm guessing it doesn't return the whole document in the case where the document is very large and that would be expensive but updating only a few attributes is cheap. Should I continue as-is with the extra db call after to get the object again or not return the object and put it on the caller to query the db again if they need to?

Reply:
Given the (potential) cost, I would probably return the `upserted_id`, which should make it cheap for the caller to find the document if needed.

Reply:
Done https://github.com/knative-portability/large-events/pull/8/commits/29bbf737717d46f7bbb64968e726fc6792c4dbb8